rdflib/graph.py

```diff
@@ -43,8 +43,8 @@
 
 logger = logging.getLogger(__name__)
 
-# Type aliases to make unpacking what's going on a little more human friendly
-ContextNode = Union[BNode, URIRef]
+# Type aliase to make unpacking what's going on a little more human friendly
+ContextNode = Union[IdentifiedNode, Literal, str, None]
```

Comment:
The only place that `ContextNode` is used is in `DatasetQuad`, and there it is optional, so the `None` here is not needed, but it also does not hurt really.

Reply:
I still think we should re-evaluate the idea here, I would say at the very least we should consider making these private, i.e. `_ContextNode`, `_DatasetQuad` - if we use the same type aliases many times in many places we can look at making them public, but even then I would say they should rather go into `rdflib.typing` so we can reserve the names in here if we later on want to make them abstract base classes or real classes.

Reply:
Also, the problem here is, this is used in a single place, and in that place ContextNode will never be a string, but always be a `Node` I think:

https://github.com/RDFLib/rdflib/blob/4eba1a098698f301e2996fba23a72927fb5d07a2/rdflib/graph.py#L2036-L2041

Because if it is not a node it will be converted to URIRef here:

https://github.com/RDFLib/rdflib/blob/4eba1a098698f301e2996fba23a72927fb5d07a2/rdflib/graph.py#L342-L345

And actually it should also never be anything other than the Default Graph (in this case identified by none), an IRI or a blank node:

https://www.w3.org/TR/rdf11-concepts/#section-dataset

> 
> An RDF dataset is a collection of [RDF graphs](https://www.w3.org/TR/rdf11-concepts/#dfn-rdf-graph), and comprises:
> 
>  - Exactly one default graph, being an [RDF graph](https://www.w3.org/TR/rdf11-concepts/#dfn-rdf-graph). The default graph does not have a name and may be empty.
>  - Zero or more named graphs. Each named graph is a pair consisting of an [IRI](https://www.w3.org/TR/rdf11-concepts/#dfn-iri) or a blank node (the graph name), and an [RDF graph](https://www.w3.org/TR/rdf11-concepts/#dfn-rdf-graph). Graph names are unique within an RDF dataset.